morphia/src/main/java/org/mongodb/morphia/DatastoreImpl.java

```diff
@@ -344,11 +344,16 @@ void process(final MappedClass mc, final Validation validation) {
         }
 
         updateForVersioning(query, operations);
-        DBObject res = dbColl.findAndModify(query.getQueryObject(), options.copy()
-                                                                           .sort(query.getSortObject())
-                                                                           .projection(query.getFieldsObject())
-                                                                           .update(((UpdateOpsImpl<T>) operations).getOps())
-                                           .getOptions());
+        DBObject queryObject = query.getQueryObject();
+        if (operations.isIsolated()) {
```

Comment:
My reading of $isolated is that it can safely be ignored for findAndModify, as that command only ever updates a single document.